tests/osaka/eip7934_block_rlp_limit/test_max_block_rlp_size.py

```diff
@@ -0,0 +1,129 @@
+"""
+abstract: Test [EIP-7934: RLP Execution Block Size Limit](https://eips.ethereum.org/EIPS/eip-7934)
+    Tests for [EIP-7934: RLP Execution Block Size Limit](https://eips.ethereum.org/EIPS/eip-7934).
+"""
+
+from typing import List
+
+import pytest
+
+from ethereum_test_forks import Fork
+from ethereum_test_tools import (
+    Alloc,
+    Block,
+    BlockchainTestFiller,
+    BlockException,
+    Bytes,
+    Transaction,
+)
+
+from .spec import Spec, ref_spec_7934
+
+REFERENCE_SPEC_GIT_PATH = ref_spec_7934.git_path
+REFERENCE_SPEC_VERSION = ref_spec_7934.version
+
+pytestmark = pytest.mark.valid_from("Osaka")  # TODO: should this be Prague?
+
+
+@pytest.fixture
+def block_size_limit(fork: Fork) -> int:
+    """Get the fork-specific block RLP size limit."""
+    limit = fork.block_rlp_size_limit()
+    if limit is None:
+        raise ValueError("Fork does not implement block RLP size limit")
+    assert limit == Spec.MAX_RLP_BLOCK_SIZE, (
+        f"Expected block RLP size limit to be {Spec.MAX_RLP_BLOCK_SIZE}, "
+        f"but got {limit} for fork {fork.name}"
+    )
+    return limit
+
+
+@pytest.fixture
+def block_errors() -> List[BlockException]:
+    """Block exceptions expected for blocks that exceed the `MAX_RLP_BLOCK_SIZE`."""
+    return [BlockException.RLP_BLOCK_LIMIT_EXCEEDED]
+
+
+@pytest.fixture
+def oversized_calldata(block_size_limit: int) -> Bytes:
+    """Generate oversized calldata that exceeds the block limit."""
+    size = block_size_limit + 10000  # Add buffer to exceed the limit
+    return Bytes(b"0x" + b"ff" * size)
+
+
+@pytest.fixture
+def large_calldata(block_size_limit: int) -> Bytes:
+    """Generate large calldata that's about 1/4 of the block limit."""
+    size = block_size_limit // 4
```

Comment:
I think we need to query the current transaction gas limit (30M due to https://github.com/ethereum/EIPs/blob/master/EIPS/eip-7825.md), and from that calculate the maximum size of calldata that a single transaction can have.